src/otadata.toit

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2023 Toitware ApS.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+import binary show LITTLE_ENDIAN
+import crypto.crc show Crc
+
+class Otadata:
+  static ENTRY_OFFSET_0 ::= 0
+  static ENTRY_OFFSET_1 ::= 4096
+  static OTADATA_SIZE ::= 8192
+  select_entries/List
+
+  constructor .select_entries:
+
+  static decode bytes/ByteArray -> Otadata:
+    select_entry1 := SelectEntry.decode bytes --offset=ENTRY_OFFSET_0
+    select_entry2 := SelectEntry.decode bytes --offset=ENTRY_OFFSET_1
+    return Otadata [select_entry1, select_entry2]
+
+  select_entry1 -> SelectEntry:
+    return select_entries[0]
+
+  select_entry2 -> SelectEntry:
+    return select_entries[1]
+
+  encode -> ByteArray:
+    bytes := ByteArray OTADATA_SIZE: -1
+    bytes.replace ENTRY_OFFSET_0 select_entry1.encode
+    bytes.replace ENTRY_OFFSET_1 select_entry2.encode
+    return bytes
+
+class SelectEntry:
+  /**
+  Monitor the first boot.
+  In the bootloader this state is changed to $STATE_IMAGE_PENDING_VERIFY
+  */
+  static STATE_IMAGE_NEW ::= 0
+  /**
+  First time this image has been booted.
+  The bootloader changes this state to $STATE_IMAGE_ABORTED.
+  */
+  static STATE_IMAGE_PENDING_VERIFY ::= 1
+  /**
+  The image has been marked as workable.
```

Comment:
"working"?  (Here and other places.)

Reply:
The docs seem to use "workable" from time to time. But they aren't native speakers either...
done.